catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java

```diff
@@ -219,18 +219,38 @@ QueryResponse query(
       queryResponse = processPostQueryAccessPlugins(queryResponse);
       queryResponse = processPostQueryPlugins(queryResponse);
 
-      LOGGER.trace("AfterPostQueryFilter result size: {}", queryResponse.getResults().size());
-      LOGGER.trace("Total Hit count: {}", queryResponse.getHits());
+      log(queryResponse);
 
     } catch (OAuthPluginException e) {
       throw e;
     } catch (RuntimeException re) {
+      LOGGER.debug("Unhandled runtime exception during query", re);
       throw new UnsupportedQueryException("Exception during runtime while performing query", re);
     }
 
     return queryResponse;
   }
 
+  private void log(QueryResponse queryResponse) {
+    LOGGER.trace("AfterPostQueryFilter result size: {}", queryResponse.getResults().size());
+    LOGGER.trace("Total Hit count: {}", queryResponse.getHits());
+    if (LOGGER.isTraceEnabled() && queryResponse.getProcessingDetails() != null) {
+      LOGGER.trace("Processing details count: {}", queryResponse.getProcessingDetails().size());
+      for (ProcessingDetails details : queryResponse.getProcessingDetails()) {
+        if (details.getException() != null) {
+          LOGGER.trace(
+              "Source [{}] query exception", details.getSourceId(), details.getException());
+        }
+        if (details.getWarnings() != null && details.getWarnings().size() > 0) {
```

Comment:
✏️ `!isEmpty()` instead of size check.

Reply:
What are the trade offs between a size check and using `isEmpty`?

https://github.com/AdoptOpenJDK/openjdk-jdk8u/blob/master/jdk/src/share/classes/java/util/ArrayList.java#L278-L294

Reply:
Clarity over performance (which is negligible). Having been tasked with resolving a lot of SonarQube findings recently, it's fresh on my mind. Though I can't find a link to it right now, it pops up as a minor code smell.

Reply:
I think I found it.

https://rules.sonarsource.com/java/tag/clumsy/RSPEC-1155